### PR TITLE
Make e2e test itself in github action

### PR DESCRIPTION
**Pull request by chunriyeqiongsaigao**
### This PR introduces several directories and files, and their relationships are depicted as follows:
- skywalking-infra-e2e/
    - .github/workflows/
        - **e2e-test.yaml**
    - test/**e2e**/
        - **e2e.yaml**
        - **docker-compose.yaml**
        - **fail-fast**/
            - **verify.yaml**
            - **expected.yaml**
            - **internal**/
                - **expected.yaml**
        - **non-fail-fast**/
            - **verify.yaml**
            - **expected.yaml**
            - **internal**/
                - **expected.yaml**

### This PR currently implements the `fail-fast` and `non-fail-fast` testing modes in the `non-concurrency` mode only. The plan is to implement the `fail-fast` and `non-fail-fast` testing modes in the `concurrency` mode in the subsequent PRs.

### CODE REVIEW

File: .github/workflows/e2e-test.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+name: e2e-test
+
+on:
+  pull_request:
+
+jobs:
+  build:
+    
+    name: Build
```
Comment (wu-sheng) [inline .github/workflows/e2e-test.yaml:27]: Should this rename to e2e? Build seems strange.

Comment (chunriyeqiongsaigao) [reply .github/workflows/e2e-test.yaml:27]: OK, Thanks very much !

File: .github/workflows/e2e-test.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        id: go
+
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+
+      - name: Build e2e
+        run: make linux
+
+      - name: Run E2E Test
+        uses: apache/skywalking-infra-e2e@main
```
Comment (kezhenxu94) [inline .github/workflows/e2e-test.yaml:43]: This always run the codes on main branch, in order to run the codes in the PR, not on the main branch, maybe you should try something like:

```suggestion
        uses: .
```

**Review by fgksgf — COMMENTED** [review]
Currently, `<feature>/expected.yaml` and `<feature>/internal/expected.yaml` seems a bit confusing, how about exchange them and move `<feature>/verify.yaml` to `<feature>/internal/verify.yaml`?////////////////

File: .github/workflows/e2e-test.yaml (hunk 2 above)
Comment (fgksgf) [reply .github/workflows/e2e-test.yaml:43]: `uses: apache/skywalking-infra-e2e@main` is just what we want, it's the external e2e, and note that we build a e2e using the PR code as the internal e2e (to be tested)

File: .github/workflows/e2e-test.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+name: E2E Test
+
+on:
+  pull_request:
```
[the remark this answers is not in the document]
Comment (fgksgf) [reply .github/workflows/e2e-test.yaml:22]: I think we can add this to ensure only one e2e workflow is running for each PR:
```
concurrency:
  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
  cancel-in-progress: true
```

[the remark this answers is not in the document]
Comment (chunriyeqiongsaigao) [reply .github/workflows/e2e-test.yaml:22]: Thanks

File: .github/workflows/e2e-test.yaml (hunk 2 above)
Comment (kezhenxu94) [reply .github/workflows/e2e-test.yaml:43]: Oh right!

**Review by fgksgf — APPROVED** [review]
LGTM, look forward to adding other test cases 👍 @chunriyeqiongsaigao

**Pull request merged**